url29ff2e53f3_nicolas3470_Settlers_of_Catan_AI_tgz-pJ8-robotGameJ8/src/soc/game/SOCGame.java

```diff
@@ -0,0 +1,4792 @@
+     *           and set game's minimum version by calling
+     *           {@link SOCGameOption#optionsMinimumVersion(Hashtable)}.
+     * @throws IllegalArgumentException if op contains unknown options, or any
+     *             object class besides {@link SOCGameOption}, or if game name
+     *             fails {@link SOCMessage#isSingleLineAndSafe(String)}.
+     * @since 1.1.07
+     */
+    public SOCGame(String n, boolean a, Hashtable op)
+        throws IllegalArgumentException
+    {
+        // For places to initialize fields, see also resetAsCopy().
+
+        if (! SOCMessage.isSingleLineAndSafe(n))
+            throw new IllegalArgumentException("n");
+
+        active = a;
+        inUse = false;
+        name = n;
+        if (op != null)
+        {
+           SOCGameOption maxpl = (SOCGameOption) op.get("PL");
+           if ((maxpl != null) && (maxpl.getIntValue() > 4))
+               maxPlayers = MAXPLAYERS;  // == 6
+           else
+               maxPlayers = 4;
+        } else {
+            maxPlayers = 4;
+        }
+        board = new SOCBoard(op, maxPlayers);
+        players = new SOCPlayer[maxPlayers];
+        seats = new int[maxPlayers];
+        seatLocks = new boolean[maxPlayers];
+        boardResetVotes = new int[maxPlayers];
+
+        for (int i = 0; i < maxPlayers; i++)
+        {
+            players[i] = new SOCPlayer(i, this);
+            seats[i] = VACANT;
+            seatLocks[i] = UNLOCKED;
+        }
+
+        currentPlayerNumber = -1;
+        firstPlayerNumber = -1;
+        currentDice = -1;
+        playerWithLargestArmy = -1;
+        playerWithLongestRoad = -1;
+        boardResetVoteRequester = -1;
+        playerWithWin = -1;
+        gameState = NEW;
+        turnCount = 0;
+        roundCount = 0;
+        forcingEndTurn = false;
+        askedSpecialBuildPhase = false;
+        placingRobberForKnightCard = false;
+        oldPlayerWithLongestRoad = new Stack();
+
+        opts = op;
+        if (op == null)
+        {
+            clientVersionMinRequired = -1;
+        } else {
+            if (! SOCGameOption.adjustOptionsToKnown(op, null))
+                throw new IllegalArgumentException("op: unknown option");
+
+            // the adjust method will also throw IllegalArg if a non-SOCGameOption
+            // object is found within opts.
+
+            clientVersionMinRequired = SOCGameOption.optionsMinimumVersion(op);
+        }
+
+        if (maxPlayers > 4)
+            numDevCards = NUM_DEVCARDS_6PLAYER;
+        else
+            numDevCards = NUM_DEVCARDS_STANDARD;
+
+        if (active)
+            startTime = new Date();
+    }
+
+    /**
+     * take the monitor for this game
+     */
+    public synchronized void takeMonitor()
+    {
+        //D.ebugPrintln("TAKE MONITOR");
+        while (inUse)
+        {
+            try
+            {
+                wait(1000);
+            }
+            catch (InterruptedException e)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)